feg/gateway/services/n7_n40_proxy/n7/model_conversion.go

```diff
@@ -0,0 +1,595 @@
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package n7
+
+import (
+	b64 "encoding/base64"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/golang/protobuf/ptypes/timestamp"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	"magma/feg/gateway/policydb"
+	common5g "magma/feg/gateway/sbi/specs/TS29122CommonData"
+	n7_sbi "magma/feg/gateway/sbi/specs/TS29512NpcfSMPolicyControl"
+	sbi "magma/feg/gateway/sbi/specs/TS29571CommonData"
+	"magma/lte/cloud/go/protos"
+)
+
+//
+// From Proto to SBI types
+//
+
+func GetSmPolicyContextDataN7(
+	request *protos.CreateSessionRequest,
+	notifyApiRoot string,
+) *n7_sbi.PostSmPoliciesJSONRequestBody {
+	imsi := RemoveIMSIPrefix(request.GetCommonContext().GetSid().Id)
+	common := request.GetCommonContext()
+	ratType := common.GetRatType()
+
+	reqBody := &n7_sbi.PostSmPoliciesJSONRequestBody{
+		Supi:              sbi.Supi(imsi),
+		Ipv4Address:       getSbiIpv4(common.GetUeIpv4()),
+		Ipv6AddressPrefix: getSbiIpv6(common.GetUeIpv6()),
+		Dnn:               sbi.Dnn(common.GetApn()),
+		Gpsi:              getSbiGpsi(string(common.GetMsisdn())),
+		RatType:           getSbiRatType(ratType),
+		AccessType:        getSbiAccessType(ratType),
+		UeTimeZone:        GetSbiTimeZone(request.GetAccessTimezone()),
+		NotificationUri:   GenNotifyUrl(notifyApiRoot, request.SessionId),
+	}
+	if request.RatSpecificContext != nil {
+		ratSpecific := request.GetRatSpecificContext().GetContext()
+		switch context := ratSpecific.(type) {
+		case *protos.RatSpecificContext_M5GsmSessionContext:
+			m5gCtx := context.M5GsmSessionContext
+			reqBody.PduSessionId = sbi.PduSessionId(m5gCtx.GetPduSessionId())
+			reqBody.Gpsi = getSbiGpsi(m5gCtx.GetGpsi())
+			reqBody.PduSessionType = getSbiPduSesionType(m5gCtx.GetPduSessionType())
+		}
+	}
+
+	return reqBody
+}
+
+func getSbiRatType(ratType protos.RATType) *sbi.RatType {
+	var sbiRatType sbi.RatType
+	switch ratType {
+	case protos.RATType_TGPP_LTE:
+		sbiRatType = sbi.RatTypeEUTRA
+	case protos.RATType_TGPP_NR:
+		sbiRatType = sbi.RatTypeNR
+	case protos.RATType_TGPP_WLAN:
+		sbiRatType = sbi.RatTypeWLAN
+	default:
+		return nil
+	}
+	return &sbiRatType
+}
+
+func getSbiAccessType(ratType protos.RATType) *sbi.AccessType {
+	var accessType sbi.AccessType
+	switch ratType {
+	case protos.RATType_TGPP_LTE:
+		accessType = sbi.AccessTypeN3GPPACCESS
+	case protos.RATType_TGPP_NR:
+		accessType = sbi.AccessTypeN3GPPACCESS
+	case protos.RATType_TGPP_WLAN:
+		accessType = sbi.AccessTypeNON3GPPACCESS
+	default:
+		return nil
+	}
+	return &accessType
+}
+
+func GetSbiTimeZone(tzProto *protos.Timezone) *sbi.TimeZone {
```

Comment:
btw, I haven't check the scope of all those functions, but note that if you are not going to use those functions outside `package n7` you do not need to capitalize any (meaning they can be private)

Note normally public methods normally require a definition/doc string (not necessary here). So the more private the less description go will require.

In this case, just make sure you really need to export those functions or not.